pandas/io/sql.py

```diff
@@ -1248,19 +1248,23 @@ def to_sql(self, frame, name, if_exists='fail', index=True,
                          schema=schema, dtype=dtype)
         table.create()
         table.insert(chunksize)
-        # check for potentially case sensitivity issues (GH7815)
-        engine = self.connectable.engine
-        with self.connectable.connect() as conn:
-            table_names = engine.table_names(
-                schema=schema or self.meta.schema,
-                connection=conn,
-            )
-        if name not in table_names:
-            warnings.warn("The provided table name '{0}' is not found exactly "
-                          "as such in the database after writing the table, "
-                          "possibly due to case sensitivity issues. Consider "
-                          "using lower case table names.".format(name),
-                          UserWarning)
+        if (not name.isdigit() and not name.islower()):
+            # check for potentially case sensitivity issues (GH7815)
+            # Only check when name is not a number and name is not lower case
+            engine = self.connectable.engine
+            with self.connectable.connect() as conn:
+                table_names = engine.table_names(
```

Comment:
Well, I indeed put a comment here a while ago that I couldn't test it .. :-) 
And I still think this is the case, as we would have to change the settings of the postgresql table on travis (to have this situation occurring), and not sure how to do this/if this is worth the trouble